pkg/logs/metrics/metrics.go

```diff
@@ -58,6 +58,11 @@ var (
 	// TlmEncodedBytesSent is the total number of sent bytes after encoding if any
 	TlmEncodedBytesSent = telemetry.NewCounter("logs", "encoded_bytes_sent",
 		nil, "Total number of sent bytes after encoding if any")
+	// BytesDropped is the number of bytes lost before they could be consumed by the agent, such as after a log rotation
+	BytesDropped = expvar.Int{}
+	// TlmBytesDropped is the number of bytes lost before they could be consumed by the agent, such as after log rotation
+	TlmBytesDropped = telemetry.NewCounter("logs", "bytes_dropped",
```

Comment:
NIT: I wonder if we should call this `bytes_missed` instead of dropped. 
We already have the concept of dropped logs/payloads where the agent explicitly throws them away. In this case the agent doesn't have a chance to tail them. 

(Open to opinions here)

Reply:
Two thumbs up, I like that better